src/Features/CSharp/Portable/EditAndContinue/StatementSyntaxComparer.cs

```diff
@@ -89,24 +90,59 @@ private IEnumerable<SyntaxNode> EnumerateRootChildren(SyntaxNode root)
         {
             RoslynDebug.Assert(_oldRoot != null);
             RoslynDebug.Assert(_newRoot != null);
-            RoslynDebug.Assert(_oldRootChild != null);
-            RoslynDebug.Assert(_newRootChild != null);
-
-            var child = (root == _oldRoot) ? _oldRootChild : _newRootChild;
 
-            if (GetLabelImpl(child) != Label.Ignored)
+            foreach (var child in GetRootChildren(root))
             {
-                yield return child;
+                if (GetLabelImpl(child) != Label.Ignored)
+                {
+                    yield return child;
+                }
+                else
+                {
+                    foreach (var descendant in child.DescendantNodes(DescendIntoChildren))
+                    {
+                        if (HasLabel(descendant))
+                        {
+                            yield return descendant;
+                        }
+                    }
+                }
             }
-            else
+        }
+
+        private IEnumerable<SyntaxNode> GetRootChildren(SyntaxNode node)
```

Comment:
>GetRootChildren [](start = 40, length = 15)

Looks good. I'd just rename this method to `EnumerateChildren` - GetRootChildren suggests that node is a root. But it is not expected to be a root.
Also, can we now merge ENumerateNonRootChildren and EnumerateRootChildren? Seems like we have one extra layer there.